src/CP/valueselection/learning/rewards/generalreward.jl

```diff
@@ -0,0 +1,122 @@
+"""
+    struct GeneralReward <: AbstractReward end
+
+This is a general reward encouraging a smart exploration of the tree in the case DFSearch is used.
```

Comment:
I think it would be nice to add some comments on the general design of the reward, in the same way as it was done in the description of the pull request.

Like for example that 
> By design, we make sure that any found feasible solution always gets a higher reward than any infeasible solution.

Reply:
Very good idea to improve clarity. Done.